### DISCUSSION TITLE
One frequency fails, the rest are fine

### DISCUSSION THREAD

**Discussion by wvthoog** [rtl-airband/RTLSDR-Airband#242] [Q&A]
Hi,

so this is driving me nuts since I've been tackling this problem for weeks now. I've tried so many different configuration options and versions of the software that i'm asking here in the hope of finding out what's wrong.

The hardware that i'm using:
- Orange Pi Zero - Allwinner H2 / 512Mb RAM / USB2
- RTL-SDR - RTL2838
- Nagoya NA-771 - 144/430Mhz antenna
- Connected with Ethernet (100Mb/s)

The plan was to scan for frequenties in the marine band and publish them to Broadcastify. This works great except for one channel which happens to be the most active and relevant one. 

When i test it using the same dongle (in Ubuntu) and fire up GQRX the 'problem' channel (161.575Mhz) comes in nice and crisp. It helps when i use LNA but a adequate reception can also be achieved without it if i raise the gain a little bit.  

Mainly tested with RTL-Airband version 3 but recently also gave version 4 a go. Both give the same result. If i had to describe the problem the channel seams to weak and has a lot of static. Raising the gain to full (49.6db) some speech is recognizable but just barely due to a lot of static noise. Thought maybe the PPM value was of and ran "rtl_test -p" for several minutes which gave a value of 1. So that might not be the case, also because all the other signals are received just fine. 

This is the latest (relevant) portion of my configuration file
```
#fft_size = 1024;
#fft_size = 512;

devices: ({
#  type = "soapysdr";
#  device_string = "driver=rtlsdr,rtl=0";
#  device_string = "driver=rtlsdr,rtl=0,buffers=20";
#  device_string = "driver=rtlsdr,rtl=0,digital_agc=true";
#  device_string = "driver=rtlsdr,,rfgain_sel=5,if_mode=Zero-IF,iqcorr_ctrl=true,extref_ctrl=false,biasT_ctrl=false,rfnotch_ctrl=yes";
  type = "rtlsdr";
  index = 0;
#  gain = 49.6;
  gain = 50;
#  gain = " PGA=10, LNA=49, TIA=5 ";
#  gain = " LNA=49.6 ";
#  gain = "LNA=16";
#  gain = "LNA=15,MIX=15,VGA=5.5";
#  gain = "LNA=15,MIX=15,VGA=15,TIA=15,PGA=15,IFGR=15";
#  gain = "IFGR=34,RFGR=4";
  correction = 2;
  mode = "scan";
#  sample_rate = 2.4;
#  sample_rate = 2.048;
#  sample_rate = 1.792;
#  sample_rate = 1.024;
#  sample_rate = 1.8;
#  sample_rate = 2500000;
#  sample_rate = 10000000;
  antenna = "Antenna A"
#  antenna = "RX"
#  buffers = 20;
  channels:
  (
    {
      squelch_snr_threshold = 10;
#      squelch = 310;
#      squelch_threshold = -40;
#      freqs = ( 156.8, 156.2, 160.8, 157.1, 161.7, 156.575, 156.725, 156.975, 161.575 );
#      squelch = ( -1, -1, -1, -1, -1, -1, -1, -1, -1);
#      squelch = ( -1, -1, -1, 300, -1, -1, -1, -1, -1);
#      freqs = ( 156.8, 156.375, 156.975, 157.1, 161,575, 161.7, 156.575, 156.5, 156.65, 157.125, 161.725 );
#      freqs = ( 156.8, 156.375, 156.975, 161.575, 157.1, 156.575, 156.5, 156.65, 157.125, 161.725 );
#      squelch = ( 430, 420, 430, 430, 420, 420, 420, 420, 420, 430, 420 );
      freqs = ( 161.575 );
#      squelch = 410;
      modulation = "nfm";
#      lowpass = 90;
#      highpass = -1;
#      afc = 0;
#      labels = ( "Nood", "Nood Afhandeling", "Sector Dordrecht Schip", "Sector Dordrecht Wal", "Merwebrug Schip", "Merwebrug Wal", "Stadsbrug", "Interschip10", "Interschip13", "Bunkeren Schip", "Bunkeren Wal" );
      outputs: (
etc etc
```
As you can see it's a mess since i'm at a loss on how to get i clear signal from 161.575. 

Any pointers on how to possibly resolve this would be greatly appreciated.

**Comment by szpajder** [comment]
Can you show how this crisp and clean signal looks like in GQRX? For
example, take a screenshot of the spectrum and the waterfall when there is
transmission on that channel. If possible, please use the same antenna and
dongle with the same settings (gain, ppm) that you run rtl_airband on.

Have you tried to run the program with textual waterfalls enabled? What are
the typical values of noise floor and signal level when the channel is
silent and when there is transmission going on?

**Reply by wvthoog** [reply]
Thank you for the quick reply. The setup is at another location but i will be there tomorrow and make some screenshots and values from GQRX. 

I've switched back to RTL_Airband version 3 (master) and these are values when i leave fft_size and sample_rate at default (commented out) and set the gain to 49.6 dB and raise the squelch to 350 

```
 402/350* 161.575 
 368/350* 161.575 
 372/350* 161.575 
 359/350* 161.575 
 353/350  161.575 
 360/350  161.575 
 351/350  161.575 
 388/350* 161.575 
 402/350* 161.575
```

Above values when there is a transmission. Some speech is recognizable but mainly garbled noise

```
 333/350  161.575 
 354/350  161.575 
 333/350  161.575 
 330/350  161.575 
 341/350  161.575 
 348/350  161.575 
 317/350  161.575 
 338/350  161.575
```

These values when the channel is silent

**Comment by szpajder** [comment]
This looks like a massive overload. Reduce the gain, so that the noise
level in steady state is between 10 and 20 (or about -45 to -50 in case you
are running the latest unstable branch which displays signal levels in
dBFS). Then get the measurements again.

**Reply by wvthoog** [reply]
Sorry about that, ran it with one of my last configurations. Which was indeed was way too much gain. 

Have set the gain to 10 db and the squelch to 30 which yields the following result.

```
  25/ 30  161.575 
  25/ 30  161.575 
  26/ 30  161.575 
  24/ 30  161.575 
  23/ 30  161.575 
  24/ 30  161.575 
  24/ 30  161.575 
  25/ 30  161.575 
  26/ 30  161.575 
  25/ 30  161.575 
  25/ 30  161.575
```

Problem here is that there is never no 'voice' activity, only some noises (which sound like scratches) periodically. Also I've noticed that there are no strong signals. Signal barely exceeds the 30 mark and when it does (again) sounds like a scratching noise.

Also tried using gain 5 and squelch 20, which gives the values you suggested. Between 10 and 20. Basically the same result as described above.

**Comment by charlie-foxtrot** [comment]
Can you try without manually setting squelch so it uses auto squelch?

Also can you try using the “multichannel” mode instead of scan to see if that behaves the same.

**Reply by wvthoog** [reply]
Have made the changes you suggested. Commented out squelch. Centerfreq = 160 and freq = 161.575

Results:

```
  24/ 22  
  24/ 22  
  23/ 22  
  23/ 22  
  23/ 22
  24/ 22  
  24/ 22  
  22/ 22  
  24/ 22  
  23/ 22
```

Hard to say if the changes worked because it's a marine channel and they tend to be very quiet at night. I'll check tomorrow morning.

**Reply by szpajder** [reply]
Receiver bandwidth is equal to the sample rate. If you have it set to 2.4 Msps, then you can't set centerfreq to 160 because the channel would be too far away. Set it to 161.5 for example.

**Reply by wvthoog** [reply]
Have set the centerfreq to 161.575. Will report back tomorrow when there is more activity on the channel. Thank you both for the support. Really appreciate it.

**Reply by szpajder** [reply]
161.575 is not good either because the channel is then located exactly in the center of the band. If the dongle has a large DC offset (a.k.a. spike at the center), then the spike will obliterate the channel of interest and probably interfere with squelch operation. Modern receivers usually don't have this issue (or they do but it is compensated by the driver) but since you want us to solve your reception issue, then please play on the safe side and avoid settings that may introduce additional problems unrelated to the original issue. Please set the centerfreq to 161.5.

**Reply by wvthoog** [reply]
Copy that. Made the correction. Will post the results together with GQRX settings and screenshots tomorrow. Thanks again.

**Reply by wvthoog** [reply]
Well, it was a Saturday and set at location for an hour. (tested with GQRX from laptop) No communication from 161.575. Installed another antenna just to test out if it had better reception. Wouldn't make no difference. 

Tested with steps to gain 50, and even tuned LNA to 49.6. Channel seemed to be quiet. 

What i'm going to do now is bring my Baofeng radio and tune it to 161.575 and monitor GQRX on a week day.
